vital/types/bounding_box.h

```diff
@@ -53,6 +53,13 @@ class bounding_box
 public:
   typedef Eigen::Matrix< T, 2, 1 > vector_type;
 
+  /**
+   * @brief Create default (invalid) box.
+   */
+  bounding_box()
+    : bounding_box(vector_type(-1,-1), vector_type(-1, -1 ))
+  { }
```

Comment:
If there is a need for an invalid bounding box, then there should be an `is_valid()` of `valid(`) method also (whatever the prevailing convention) so it is easy to tell if it is valid. That way **valid** becomes a property of the object and users don't have to go mucking around and check against magic values.

Reply:
I think we should have a 'is_empty' method, and it will return true if all values are >0

I used -1 per https://github.com/Kitware/kwiver/blob/master/arrows/core/write_object_track_set_kw18.cxx#L108

But I think the test should be for non negative values

Reply:
unless you like is_valid, not sure on terminology for this...

Reply:
Can't predictive boxes have corners less than 0? (I think the "safe" way to implement this, assuming Eigen lets you, is to define "invalid" as having a negative width and/or height. I want to say that's how e.g. QRect[F] is implemented...)

Reply:
That sounds good. `bool is_valid() const;` will work as a signature.

Reply:
makes sense
how about

`bool is_valid()
  {
    return (width() >= 0 && height() >= 0);
  }`

Reply:
@linus-sherrill have your comments/concerns been addressed?